examples/common/tracing/decoder/logging/ToCertificateString.cpp

```diff
@@ -40,7 +40,7 @@ const char * ToCertificate(const chip::ByteSpan & source, chip::MutableCharSpan
         return destination.data();
     }
 
-    size_t base64DataLen = BASE64_ENCODED_LEN(source.size()) + 1;
+    size_t base64DataLen = BASE64_ENCODED_LEN(source.size());
```

Comment:
Isn't the +1 there needed to account for the null-terminator?

Reply:
@bzbarsky-apple It will always add a `\n` after each line or the end of the string : [ToCertificateString.cpp#L81](https://github.com/project-chip/connectedhomeip/blob/master/examples/common/tracing/decoder/logging/ToCertificateString.cpp#L81), null-terminator could indicate the end of the string, but don't need to be counted into the total length.

Otherwise, if the `\0` is the last character left, it will print an extra blank line, and cause the parsing issue.

Reply:
What I mean is that this code:
```
    str.Alloc(base64DataLen);
    auto encodedLen       = chip::Base64Encode(source.data(), source.size(), str.Get());
    str.Get()[encodedLen] = '\0';
```
will write outside the buffer if `encodedLen` ends up being `BASE64_ENCODED_LEN(source.size())`, which it can be, right?

Reply:
And also, we would write outside the `destination` buffer...

We can either not add 1 here but make sure we add 1 to the relevant size checks and allocations, or we can use separate variables for "length of the data" and "buffer needed to hold the data", or something else, but what we absolutely cannot do if overrun buffers.  That's how you get security bugs.

Reply:
Fixed accordingly, thanks!